test/shoppingcart_test.rb

```diff
@@ -0,0 +1,115 @@
+require 'minitest/autorun'
+require 'minitest/pride'
+require './lib/product'
+require './lib/shoppingcart'
+
+
+
+class Test < Minitest::Test
+
+  def test_it_exists
+    cart = ShoppingCart.new("King Soopers", "30items")
+    assert_instance_of ShoppingCart, shoppingcart
+  end
+
+  def test_it_has_a_name
+    cart = ShoppingCart.new("King Soopers", "30items")
+    assert_equal "King Soopers", cart.name
+  end
+
+  def test_it_has_a_category
+    cart = ShoppingCart.new("King Soopers", "30items")
+    assert_equal 30, cart.capacity
+  end
+
+  def test_it_starts_empty
+    cart = ShoppingCart.new("King Soopers", "30items")
+    assert_equal [], cart.products
+  end
+
+  def test_it_can_have_products
+    cart = ShoppingCart.new("King Soopers", "30items")
+    product1 = Product.new(:paper, 'toilet paper', 3.70, '10')
+    product2 = Product.new(:meat, 'chicken', 4.50, '2')
+    cart.add_product(product1)
+    cart.add_product(product2)
+    assert_equal [product1, product2], cart.products
+  end
+  #try to look up why test throws errors
+  # def test_it_has_details
+  #   cart = ShoppingCart.new("King Soopers", "30items")
+  #   name: "King Soopers", capacity: 30}, cart.details
+  # end
+
+  def test_total_number_of_products
+    cart = ShoppingCart.new("King Soopers", "30items")
+    product1 = Product.new(:paper, 'toilet paper', 3.70, '10')
+    product2 = Product.new(:meat, 'chicken', 4.50, '2')
+    product3 = Product.new(:paper, 'tissue paper', 1.25, '1')
+    cart.add_product(product1)
+    cart.add_product(product2)
+    cart.add_product(product3)
+
+    assert_equal 13, cart.test_total_number_of_products
+  end
+
+  def test_is_full?
+    cart = ShoppingCart.new("King Soopers", "30items")
+    product1 = Product.new(:paper, 'toilet paper', 3.70, '10')
+    product2 = Product.new(:meat, 'chicken', 4.50, '2')
+    product3 = Product.new(:paper, 'tissue paper', 1.25, '1')
+    cart.add_product(product1)
+    cart.add_product(product2)
+    cart.add_product(product3)
+
+    assert_equal false, cart.is_full?
+  end
+
+  def test_products_by_category
+    cart = ShoppingCart.new("King Soopers", "30items")
+    product1 = Product.new(:paper, 'toilet paper', 3.70, '10')
+    product2 = Product.new(:meat, 'chicken', 4.50, '2')
+    product3 = Product.new(:paper, 'tissue paper', 1.25, '1')
+    cart.add_product(product1)
+    cart.add_product(product2)
+    cart.add_product(product3)
+
+    assert_equal :paper, cart.products_by_category(:paper)
```

Comment:
you'd want the expected to be the array of products that have the :paper category